articles/rules-are-dangerous.md

```diff
@@ -0,0 +1,53 @@
+* The dependency inversion principle says “Abstractions should not depend on details. Details should depend on abstractions". I have seen projects that tried to follow this principle very strictly and everywhere in their software. They therefore totally avoided to use any concrete class directly as much as possible. As a result, there were a lot of interfaces with only one implementation class each and many factories to create instances of those classes. This resulted in a code base that is much more complex than needed. It makes the software difficult to understand and maintain without bringing any benefit of really needed flexibility into that software.
+* People read the book about “Clean Code” that says that many code comments can be avoided by writing self-explaining code instead. While this is true for many cases the rule is often miss-used by developers for arguing not to write any comments at all in the source code. But there are many things that you cannot explain by using good method or variable names only. E.g. why you had to implement something like that. For this purpose, good comments are still very valuable and should not be denied. I would say that a well put comment can explain much more than hundred good variable names.
+* The article by Martin Fowler about the “Page Object Pattern” once said to never put assertions into page objects. Someone read that article and wanted to convince the whole project team that it is very important to follow this rule very strictly. The arguments for the rule in the blog post were not very convincing to me and the benefit of it is at least very questionable for most applications of that pattern I have seen. But since Martin Fowler wrote this rule in his article, it counts as if it was written in the bible. This makes it difficult to argue and discuss constructively about it. No matter how many good examples you bring up in your context of application where the rule causes more troubles than bringing benefits, you will not be taken seriously. Such dogmatic discussions about such rules can cause you to lose valuable time and energy. A rule without good justification why it is so important and in what context, is not really a useful rule. No matter how famous and clever that person is who wrote it. Funny side note: in the meantime, the article by Fowler has been changed to mention both opinions on this topic and lets the option open to the reader to decide what fits best in his concrete context. Unfortunately, not all famous technical opinion makers are so wise as Mr. Fowler.
```

Comment:
```suggestion
* The article by Martin Fowler about the “Page Object Pattern” once said to never put assertions into page objects. Someone read that article and wanted to convince the whole project team that it is very important to follow this rule very strictly. The arguments for the rule in the blog post were not very convincing to me. But since Martin Fowler wrote it, it counts as if it was written in the bible. This makes it difficult to argue and discuss constructively about it. 

Such dogmatic discussions about such rules can cause you to lose valuable time and energy. A rule without good justification why it is so important and in what context, is not really a useful rule. No matter how famous and clever that person is who wrote it. 
```

Reply:
Ok, I see this has to be shortened for sure. I will maybe do it slightly different.

Reply:
Ich mach das noch - allerdings ein wenig anders. Bin mit der Kürzung nicht ganz zufrieden. Finde hier ist etwas gar viel rausgerutscht. Z.B. wär mir hier noch wichtig zu erwähnen, dass zumindest Fowler diesbezüglich den Artikel mittlerweile verbessert hat.
Auch find ich komisch dass ein Teil jetzt aus dem letzten Beispiel rausgerutscht ist, Absicht?

Reply:
Siehe anderer Vorschlag von mir

Reply:
wie besprochen durch meinen anderen Vorschlag entsprechend gelöst.